airbyte-analytics/src/main/java/io/airbyte/analytics/Deployment.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.analytics;
+
+import com.google.common.base.Preconditions;
+import io.airbyte.config.Configs;
+import io.airbyte.config.Configs.WorkerEnvironment;
+import java.util.UUID;
+
+public class Deployment {
+
+  public enum DeploymentMode {
+    OSS,
+    CLOUD
+  }
+
+  /**
+   * deployment - deployment tracking info.
+   */
+  private final DeploymentMode deploymentMode;
+  /**
+   * deploymentId - identifier for the deployment
+   */
+  private final UUID deploymentId;
```

Comment:
opinion: I think it's useful to leave a comment here summarising how deploymetId -> workspace tie together.

Reply:
Maybe `instanceId`? I think `instance` has been the most common way to refer to the concept of a `deployment` (and that's also reflected in our docs). Either is fine with me, just don't want to diverge naming. Feels like it's an `AirbyteInstance` with an `instanceId` and `deploymentMode`/`deploymentEnvironment` (vars that describe the deployment of the instance with that id).

Reply:
good call. added.

Reply:
discussed with @jrhizor offline. our goal here is to make sure the data model matches naming in the docs. we are keeping deployment, but i updated naming in docs to match deployment.